pkg/util/kubernetes/clustername/clustername.go

```diff
@@ -94,8 +95,13 @@ func getClusterName(ctx context.Context, data *clusterNameData, hostname string)
 					// try the next cloud provider
 					continue
 				}
+				// if the clustername is valid but contains a "_" in the middle of it, we will replace it later such that
+				// to make it valid to RFC1123.
 				if clusterName != "" {
 					log.Infof("Using cluster name %s auto discovered from the %s API", clusterName, cloudProvider)
+					if strings.HasSuffix(clusterName, "-") || strings.HasSuffix(clusterName, "_") {
```

Comment:
can we use here an updated version of the regex used for RFC1123 that accepts to have `_` in the middle of the string?

Reply:
Oh, I thought we didn't want to support `_` if someone i.e. set this `DD_CLUSTER_NAME=my_name`?